codec/src/main/java/io/netty/handler/codec/base64/Base64.java

```diff
@@ -120,11 +120,10 @@ public static ByteBuf encode(
             throw new NullPointerException("dialect");
         }
 
-        int len43 = (len << 2) / 3;
-        ByteBuf dest = allocator.buffer(
-                len43 +
-                        (len % 3 > 0 ? 4 : 0) + // Account for padding
-                        (breakLines ? len43 / MAX_LINE_LENGTH : 0)).order(src.order()); // New lines
+        int len43 = len43(len);
+        ByteBuf dest = allocator.buffer((int) Math.min(Integer.MAX_VALUE, (long) len43 +
+                (len % 3 > 0 ? 4 : 0) + // Account for padding
```

Comment:
This is not correct alignement... And further the value can be increased.
I suggest extract the entire calculation of the buffer size in a separate method (without superfluous casts int<->long and overflow check) :

```java
static int getEncodeBufferSize(int len, boolean breakLines) {
    long ret = len + len / 3;
    if (breakLines) {
        ret += ret / MAX_LINE_LENGTH;
    }
    // align by 4
    ret = (ret + 3) & ~3;
    return ret < Integer.MAX_VALUE ? (int) ret : Integer.MAX_VALUE;
}
```